diff --git a/src/core/CMakeLists.txt b/src/core/CMakeLists.txt
index 50adcdd..15af653 100644
--- a/src/core/CMakeLists.txt
+++ b/src/core/CMakeLists.txt
@@ -112,10 +112,10 @@ target_link_libraries(rexcore
         simde
         tomlplusplus
         spdlog::spdlog
+        xxHash::xxhash
     PRIVATE
         disruptorplus
         utf8cpp
-        xxHash::xxhash
 )
 
 if(REXGLUE_USE_D3D12)
diff --git a/src/system/CMakeLists.txt b/src/system/CMakeLists.txt
index d1139cd..186e000 100644
--- a/src/system/CMakeLists.txt
+++ b/src/system/CMakeLists.txt
@@ -84,10 +84,10 @@ target_link_libraries(rexruntime
         spdlog::spdlog
         tomlplusplus
         SDL3::SDL3
+        xxHash::xxhash
     PRIVATE
         aes128
         mspack
-        xxHash::xxhash
 )
 
 if(REXGLUE_ENABLE_TRACY)
