diff --git a/include/rex/system/interfaces/graphics.h b/include/rex/system/interfaces/graphics.h
--- a/include/rex/system/interfaces/graphics.h
+++ b/include/rex/system/interfaces/graphics.h
@@ -148,6 +148,8 @@ struct GraphicsDrawObservation {
   uint32_t index_buffer_length = 0;
   uint32_t index_format = 0;
   uint32_t index_endianness = 0;
+  uint32_t index_reset = 0;
+  bool index_reset_enabled = false;
   uint32_t vertex_index_offset = 0;
   uint32_t vertex_index_min = 0;
   uint32_t vertex_index_max = 0;
diff --git a/src/graphics/command_processor.cpp b/src/graphics/command_processor.cpp
--- a/src/graphics/command_processor.cpp
+++ b/src/graphics/command_processor.cpp
@@ -1499,6 +1499,12 @@ bool CommandProcessor::ExecutePacketType3Draw(memory::RingBuffer* reader, uint32
           observation.index_buffer_length = uint32_t(index_buffer_info.length);
           observation.index_format = uint32_t(index_buffer_info.format);
           observation.index_endianness = uint32_t(index_buffer_info.endianness);
+          observation.index_reset =
+              register_file_->Get<reg::VGT_MULTI_PRIM_IB_RESET_INDX>()
+                  .reset_indx;
+          observation.index_reset_enabled =
+              register_file_->Get<reg::PA_SU_SC_MODE_CNTL>()
+                  .multi_prim_ib_ena;
         }
         observation.vertex_index_offset =
             register_file_->Get<reg::VGT_INDX_OFFSET>().indx_offset;
