diff --git a/include/rex/system/interfaces/graphics.h b/include/rex/system/interfaces/graphics.h
index 689645a..ecf8cba 100644
--- a/include/rex/system/interfaces/graphics.h
+++ b/include/rex/system/interfaces/graphics.h
@@ -250,6 +250,11 @@ struct GraphicsDrawObservation {
   uint32_t depth_info = 0;
   uint32_t window_scissor_tl = 0;
   uint32_t window_scissor_br = 0;
+  uint32_t viewport_xscale = 0;
+  uint32_t viewport_xoffset = 0;
+  uint32_t viewport_yscale = 0;
+  uint32_t viewport_yoffset = 0;
+  uint32_t viewport_transform_control = 0;
   uint32_t rb_modecontrol = 0;
   uint32_t viz_query_condition = 0;
   uint32_t pa_sc_viz_query = 0;
diff --git a/src/graphics/command_processor.cpp b/src/graphics/command_processor.cpp
index 349244e..5ebf112 100644
--- a/src/graphics/command_processor.cpp
+++ b/src/graphics/command_processor.cpp
@@ -1662,6 +1662,16 @@ bool CommandProcessor::ExecutePacketType3_DRAW_INDX(RingBuffer* reader,
             register_file_->Get<reg::PA_SC_WINDOW_SCISSOR_TL>().value;
         observation.window_scissor_br =
             register_file_->Get<reg::PA_SC_WINDOW_SCISSOR_BR>().value;
+        observation.viewport_xscale =
+            (*register_file_)[XE_GPU_REG_PA_CL_VPORT_XSCALE];
+        observation.viewport_xoffset =
+            (*register_file_)[XE_GPU_REG_PA_CL_VPORT_XOFFSET];
+        observation.viewport_yscale =
+            (*register_file_)[XE_GPU_REG_PA_CL_VPORT_YSCALE];
+        observation.viewport_yoffset =
+            (*register_file_)[XE_GPU_REG_PA_CL_VPORT_YOFFSET];
+        observation.viewport_transform_control =
+            register_file_->Get<reg::PA_CL_VTE_CNTL>().value;
         observation.rb_modecontrol = register_file_->Get<reg::RB_MODECONTROL>().value;
         observation.viz_query_condition = viz_query_condition;
         observation.pa_sc_viz_query = viz_query.value;
